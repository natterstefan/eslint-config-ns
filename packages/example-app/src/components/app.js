import React, { useState } from 'react'

import { Text } from './text'

export const App = () => {
  const [counter, setCounter] = useState(0)
  const headline = 'Hello eslint-config-ns'

  setInterval(() => setCounter(counter + 1), 1000)

  return (
    <>
      <h1>{headline}</h1>
      <Text />
      <p>State Counter: {counter}</p>
    </>
  )
}
