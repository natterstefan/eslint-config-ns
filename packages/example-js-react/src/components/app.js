import React, { useEffect, useState } from 'react'

import { Text } from './text'

export const App = () => {
  const [counter, setCounter] = useState(0)
  const headline = 'Hello eslint-config-ns'

  useEffect(() => {
    setInterval(() => setCounter(c => c + 1), 1000)
  }, [])

  return (
    <>
      <h1>{headline}</h1>
      <Text text="Hello World" />
      <p>State Counter: {counter}</p>
    </>
  )
}
