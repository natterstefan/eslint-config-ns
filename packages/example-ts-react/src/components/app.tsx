import React, { useState, FunctionComponent } from 'react'

import { Text } from './text'

export const App: FunctionComponent = () => {
  const [counter, setCounter] = useState(0)
  const headline = 'Hello eslint-config-ns-ts'

  setInterval(() => setCounter(counter + 1), 1000)

  return (
    <>
      <h1>{headline}</h1>
      <Text text="Hello World" />
      <p>State Counter: {counter}</p>
    </>
  )
}
