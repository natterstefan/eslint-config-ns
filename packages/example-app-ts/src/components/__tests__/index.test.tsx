import React from 'react'
import { shallow } from 'enzyme'

import { App } from '../app'

jest.useFakeTimers()

describe('App', () => {
  it('renders', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.text()).toStrictEqual(
      'Hello eslint-config-ns-ts<Text />State Counter: 0',
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('increases counter every 1 second', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.text()).toStrictEqual(
      'Hello eslint-config-ns-ts<Text />State Counter: 0',
    )

    jest.advanceTimersByTime(1000)
    expect(wrapper.text()).toContain('State Counter: 1')

    jest.advanceTimersByTime(1000)
    expect(wrapper.text()).toContain('State Counter: 2')
  })
})
