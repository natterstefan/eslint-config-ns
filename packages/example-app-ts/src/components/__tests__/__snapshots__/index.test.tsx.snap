// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<Fragment>
  <h1>
    Hello eslint-config-ns-ts
  </h1>
  <Text
    text="Hello World"
  />
  <p>
    State Counter: 
    0
  </p>
</Fragment>
`;
